### PR TITLE
Add a commented copy of OLED and rotary code in default Kyria

### PR DESCRIPTION
**Pull request by precondition**
## Description

The OLED and rotary encoder code was moved from the contents of the `keyboards/splitkb/kyria/keymaps/default/keymap.c` file to `keyboards/splitkb/kyria/rev1/rev1.c` in order to keep the default functionality of these hardware features for online QMK configurator users but that move is confusing many Kyria users who can't find where is the code for the OLEDs and rotary encoders so that they can modify it.

Some find the source in `rev1/rev1.c` but they often end up editing this keyboard-level file instead of overriding the functions in their keymap. This is not the way it should be done. A keyboard user shouldn't feel compelled to change keyboard level files in order to fit with their newly made keymap.

My proposed solution is to copy the code from `rev1/rev1.c` without the `__weak__` attribute into the default `keymap.c`
, comment it all out and add an explanatory notice on top. 

I can guess that some of you won't look at the code duplication with a favorable eye. In which case, I can remove the commented out code and only keep the explanatory note of where to find the OLED and rotary functions + instruction on how to properly override them but I'm worried that too many people would find it too bothersome to copy over code, delete all occurrences of `__attribute__((weak)) ` when they can just edit `rev1/rev1.c` directly.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by splitkb** [comment]
I think it's a good idea. It's something that regularly comes up on our Discord server, and can confuse users.

File: keyboards/splitkb/kyria/keymaps/default/keymap.c [hunk 1]
```diff
@@ -215,3 +215,103 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+            0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,  1,  3,  7, 15, 14, 30, 28, 60, 56,120,112,112,112,224,224,224,224,224,224,224,224,224,224,224,224,224,224,224,224,112,112,112,120, 56, 60, 28, 30, 14, 15,  7,  3,  1,  1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0
+        };
+        // clang-format on
+        oled_write_raw_P(kyria_logo, sizeof(kyria_logo));
+    }
+}
+#endif
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
```
Comment (daskygit) [inline keyboards/splitkb/kyria/keymaps/default/keymap.c:294]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/splitkb/kyria/keymaps/default/keymap.c [hunk 2]
```diff
@@ -215,3 +215,103 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+        }
+    } else if (index == 1) {
+        // Page up/Page down
+        if (clockwise) {
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    }
+    return true;
```
Comment (daskygit) [inline keyboards/splitkb/kyria/keymaps/default/keymap.c:314]:
```suggestion
    return false; //returning false disables the keyboard level encoder bindings.
```

Returning true will also trigger the the tap codes in the kb function when using the user function.

File: keyboards/splitkb/kyria/keymaps/default/keymap.c [hunk 3]
```diff
@@ -215,3 +215,103 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+        oled_write_raw_P(kyria_logo, sizeof(kyria_logo));
+    }
+}
+#endif
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) {
+        return false;
+    }
```
Comment (drashna) [inline keyboards/splitkb/kyria/keymaps/default/keymap.c:297]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/splitkb/kyria/keymaps/default/keymap.c (hunk 2 above)
Comment (daskygit) [inline keyboards/splitkb/kyria/keymaps/default/keymap.c:314]:
```suggestion
    return false;
```
Managed to delete the wrong comment, sorry for the spam. This should return false so it doesn't run the keyboard function.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**